libraries/conf.rb

```diff
@@ -98,7 +98,7 @@ def whyrun_supported?
         execute "#{r.name}.#{r.conf_type}-systemd-reload" do
           command 'systemctl daemon-reload'
           action :nothing
-          only_if { r.is_a?(Chef::Resource::SystemdUnit) }
+          only_if { r.is_a?(Chef::Resource::SystemdUnit) && r.auto_reload }
           subscribes :run, "file[#{conf_path}]", :immediately
```

Comment:
i went ahead and made both available (`daemon_reload_*` recipes), and users can choose. there's nothing inherent to having the event handler in a recipe in the cookbook that automatically breaks chef < 12.5, so there's no version requirement introduced unless the user opts into it by including the recipe.